Description: Improve Makefile
 Make sure to use all the build flags that can be expected in the
 environment; use the correct variable for each compiler flag.
Author: Andrea Bolognani <eof@kiyuko.org>
Last-Update: 2011-11-11
Index: scrotwm.git/linux/Makefile
===================================================================
--- scrotwm.git.orig/linux/Makefile	2011-11-11 09:51:05.817058791 +0100
+++ scrotwm.git/linux/Makefile	2011-11-11 10:01:31.957058867 +0100
@@ -1,17 +1,15 @@
 # $scrotwm: Makefile,v 1.6 2011/08/10 13:44:36 marco Exp $ 
 
-CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -I.
-CFLAGS+= -DSWM_LIB=\"$(LIBDIR)/libswmhack.so.$(LVERS)\"
-LDADD+= -lX11 -lXrandr -lXtst
-
-PREFIX?= /usr/local
-BINDIR?= $(PREFIX)/bin
-LIBDIR?= $(PREFIX)/lib
-MANDIR?= $(PREFIX)/share/man
+CFLAGS += -Wall -ggdb -I.
+CPPFLAGS += -D_GNU_SOURCE -DSWM_LIB=\"$(LIBDIR)/libswmhack.so.$(LVERS)\"
+LDLIBS += -lX11 -lXrandr -lXtst
+
+PREFIX ?= /usr/local
+BINDIR ?= $(PREFIX)/bin
+LIBDIR ?= $(PREFIX)/lib
+MANDIR ?= $(PREFIX)/share/man
 
-CC= gcc
-
-LVERS= $(shell . ../lib/shlib_version; echo $$major.$$minor)
+LVERS := $(shell . ../lib/shlib_version; echo $$major.$$minor)
 
 all: scrotwm libswmhack.so.$(LVERS)
 
@@ -22,13 +20,13 @@
 	ln -sf ../lib/swm_hack.c
 
 scrotwm: scrotwm.o linux.o
-	$(CC) $(LDFLAGS) -o $@ $+ $(LDADD)
+	$(CC) $(LDFLAGS) $+ -o $@ $(LDLIBS)
 
 %.so: %.c
-	$(CC) $(CFLAGS) -c -fpic -DPIC $+ -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c -fpic -DPIC $+ -o $@
 
 libswmhack.so.$(LVERS): swm_hack.so
-	$(CC) -shared -fpic -o libswmhack.so.$(LVERS) swm_hack.so $(LDADD)
+	$(CC) $(LDFLAGS) -shared -fpic $+ -o $@ $(LDLIBS)
 
 install: all
 	install -m 755 -d $(DESTDIR)$(BINDIR)
